luncheon/shared.lua

```diff
@@ -3,25 +3,119 @@ local Headers = require "luncheon.headers"
 local utils = require "luncheon.utils"
 local log = require "log"
 
----@enum Mode
---- Enumeration of the 2 sources of data
-local Mode = {
-  --- This is a Request or Response constructed via `source` it will rely on pulling
-  --- new bytes from the `_source` function
-  Incoming = "incoming",
-  --- This is a Request or Response constructed via `new` it will assume all the
-  --- properties have been filled manually
-  Outgoing = "outgoing",
-}
```

Comment:
I worry about this being removed, I am not as confident that `_source` being present is enough but I could be wrong?

Reply:
When I looked closer at the shared code than prior reviews, this stood out to me as something that was weird to think about and use as terminology; it also seemed potentially unnecessary. Going through the usages, the presence of `_source` was enough. The way everything is setup either the Request/Response is coming entirely from a source or from user constructed arguments; I cant come up with a use case where they aren't mutally exclusive, so I think you're wrong and that mode is unnecessary.